python/ray/util/spark/start_ray_node.py

```diff
@@ -41,6 +47,9 @@
         raise ValueError("Please explicitly set --temp-dir option.")
 
     temp_dir = os.path.normpath(temp_dir)
+    # Clean up the temp dir for global mode
+    if global_mode_enabled() and os.path.exists(temp_dir):
+        shutil.rmtree(temp_dir, ignore_errors=True)
```

Comment:
Because in spark worker, multiple ray worker nodes might be launched simultaneously on the same spark worker node, these lines code might
cause race condition: one ray worker node launches and write some temp data, then they are deleted by another `start_ray_node` process.

Reply:
Ah we shouldn't clean up it for worker nodes, they're cleaned up when exiting global ray cluster